### PR TITLE
Fixed reading the config file env for Rider

### LINKED ISSUE #165: Rider: setting MIRRORD_CONFIG_FILE does not work
### Bug Description

setting MIRRORD_CONFIG_FILE in the env vars of the run config is not propagated to the the extension

### Steps to Reproduce

Just set the MIRRORD_CONFIG_FILE env and start debugging

### Backtrace

_No response_

### Relevant Logs

_No response_

### Your operating system and version

macos

### Local process

dotnet

### Local process version

_No response_

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by infiniteregrets**
closes #165

### CODE REVIEW

File: modules/products/rider/src/main/kotlin/com/metalbear/mirrord/products/rider/RiderLaunchSettingsExtension.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    ): Boolean {
+        return true
+    }
+
+    override fun executor(
+        project: Project,
+        environment: ExecutionEnvironment,
+        profile: LaunchSettingsJson.Profile
+    ): LaunchSettingsJson.Profile {
+        val configFromEnv = profile.environmentVariables?.get(CONFIG_ENV_NAME)
```
Comment (meowjesty) [inline modules/products/rider/src/main/kotlin/com/metalbear/mirrord/products/rider/RiderLaunchSettingsExtension.kt:31]: nit: can't this and the `apply` thingy be done without creating a new `val`?

File: modules/products/rider/src/main/kotlin/com/metalbear/mirrord/products/rider/RiderPatchCommandLineExtension.kt [hunk 2]
```diff
@@ -1,3 +1,5 @@
+@file:Suppress("UnstableApiUsage")
```
Comment (meowjesty) [inline modules/products/rider/src/main/kotlin/com/metalbear/mirrord/products/rider/RiderPatchCommandLineExtension.kt:1]: Probably a good idea to doc why we need to use this.

Like, say that to solve issue N we need to use API (a) which is unstable.

File: modules/products/rider/src/main/kotlin/com/metalbear/mirrord/products/rider/RiderLaunchSettingsExtension.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import com.intellij.execution.runners.ExecutionEnvironment
+import com.intellij.openapi.project.Project
+import com.jetbrains.rd.util.lifetime.Lifetime
+import com.jetbrains.rider.run.configurations.RiderConfigurationLaunchSettingsExtension
+import com.jetbrains.rider.run.configurations.RuntimeHotReloadRunConfigurationInfo
+import com.jetbrains.rider.run.configurations.launchSettings.LaunchSettingsJson
+import com.metalbear.mirrord.CONFIG_ENV_NAME
+import java.util.concurrent.ConcurrentHashMap
+
+class RiderLaunchSettingsExtension : RiderConfigurationLaunchSettingsExtension {
```
Comment (meowjesty) [inline modules/products/rider/src/main/kotlin/com/metalbear/mirrord/products/rider/RiderLaunchSettingsExtension.kt:12]: Could you add some docs for these methods, saying that we need them to solve issue N, get the MIRRORD_CONFIG env and stuff?

Comment (infiniteregrets) [reply modules/products/rider/src/main/kotlin/com/metalbear/mirrord/products/rider/RiderLaunchSettingsExtension.kt:12]: yes will add, thank you!

File: modules/products/rider/src/main/kotlin/com/metalbear/mirrord/products/rider/RiderLaunchSettingsExtension.kt [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import com.jetbrains.rider.run.configurations.RuntimeHotReloadRunConfigurationInfo
+import com.jetbrains.rider.run.configurations.launchSettings.LaunchSettingsJson
+import com.metalbear.mirrord.CONFIG_ENV_NAME
+import java.util.concurrent.ConcurrentHashMap
+
+// Need this extension since setting MIRRORD_CONFIG_FILE in the project configuration
+// is not visible to our current methods of getting the env.
+// Both `canExecute` and `executor` have access to the project's environment variables
+// and `canExecute` is called before `executor`, so just choosing `executor` at random
+// to get the environment variables.
```
Comment (meowjesty) [inline modules/products/rider/src/main/kotlin/com/metalbear/mirrord/products/rider/RiderLaunchSettingsExtension.kt:16]:
```suggestion
// and `canExecute` is called before `executor`, so we use `executor`
// to get the environment variables.
```

Comment (meowjesty) [reply modules/products/rider/src/main/kotlin/com/metalbear/mirrord/products/rider/RiderLaunchSettingsExtension.kt:16]: Do we need to implement both? If `canExecute` is basically doing nothing?

**Review by meowjesty — APPROVED** [review]
Do we need `canExecute` though?

If we do, then it's approved.

**Pull request merged**